File: src/com/company/sixtieths/Task51.java
package com.company.sixtieths;

import com.company.EulerTask;

public class Task51 implements EulerTask {

    @Override
    public int getAnswer() {
        for (int i = 56997; i < Integer.MAX_VALUE; i+=2){
            String numStr = String.valueOf(i);

        }
        return 0;
    }

    private boolean isPrime(long n) {
        if (n % 2 == 0) return n == 2;
        if (n == 1) return false;
        for (int i = 3; i * i < n; i += 2) {
            if (n % i == 0) return false;
        }
        return true;
    }
}

File: src/com/company/sixtieths/Task51.kt
package com.company.sixtieths

import kotlin.math.sqrt

fun getAnswer(): Int {
    for (i in 56997..Int.MAX_VALUE step 2)
        i.toChar()
    return 777
}

fun isPrime(number: Int): Boolean {
    if (number % 2 == 0) return number == 2
    if (number == 1) return false
    for (i in 3..sqrt(number.toDouble()).toInt() + 1 step 2) {
        if (number % i == 0) return false
    }
    return true
}

fun main() {
    println(isPrime(21))
    println(isPrime(23))
}